Tests/SonarScanner.MSBuild.Shim.Test/SonarScannerWrapperTests.cs

```diff
@@ -257,48 +267,72 @@ public void SonarScanner_UserSpecifiedEnvVars_OnlySONARSCANNEROPTSIsPassed()
-        {
+        public void WrapperError_Success_NoStdErr() =>
             TestWrapperErrorHandling(executeResult: true, addMessageToStdErr: false, expectedOutcome: true);
-        }
 
         [TestMethod]
         [WorkItem(202)] //SONARMSBRU-202
-        public void WrapperError_Success_StdErr()
-        {
+        public void WrapperError_Success_StdErr() =>
             TestWrapperErrorHandling(executeResult: true, addMessageToStdErr: true, expectedOutcome: true);
-        }
 
         [TestMethod]
-        public void WrapperError_Fail_NoStdErr()
-        {
+        public void WrapperError_Fail_NoStdErr() =>
             TestWrapperErrorHandling(executeResult: false, addMessageToStdErr: false, expectedOutcome: false);
-        }
 
         [TestMethod]
-        public void WrapperError_Fail_StdErr()
-        {
+        public void WrapperError_Fail_StdErr() =>
             TestWrapperErrorHandling(executeResult: false, addMessageToStdErr: true, expectedOutcome: false);
+
+        [TestMethod]
+        public void FindScannerExe_FindsCliZip_ExtractsItSuccessfully()
+        {
+            // Arrange
+            var logger = new TestLogger();
+
+            var scannerCliTestDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, new string[] { "sonar-scanner-cli-0", "sonar-scanner-0", "bin" });
+            TestUtils.CreateEmptyFile(scannerCliTestDirectory, "sonar-scanner.bat");
+
+            var scannerCliTestDirectoryRoot = Directory.GetParent(scannerCliTestDirectory).Parent.FullName;
+            var zipDestinationDir = Path.Combine(TestContext.TestDir, "sonar-scanner-cli-0.zip");
+            ZipFile.CreateFromDirectory(scannerCliTestDirectoryRoot, zipDestinationDir);
+
+            // Act
+            SonarScannerWrapper.FindScannerExe(logger, TestContext.TestDir, "0");
+
+            // Assert
+            Directory.Exists(Path.Combine(TestContext.TestDir, "sonar-scanner-0"));
+            logger.AssertInfoMessageExists("Unzipping sonar-scanner-cli-0.zip");
         }
 
-        private void TestWrapperErrorHandling(bool executeResult, bool addMessageToStdErr, bool expectedOutcome)
+        [TestMethod]
+        public void FindScannerExe_ZipIsAlreadyExtracted_DoesNotExtractAgain()
         {
             // Arrange
             var logger = new TestLogger();
-            var mockRunner = new MockProcessRunner(executeResult);
+            var scannerCliTestDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, new string[] { "sonar-scanner-0", "bin" });
+            TestUtils.CreateEmptyFile(scannerCliTestDirectory, "sonar-scanner.bat");
+            var scannerCliTestDirectoryRoot = Directory.GetParent(scannerCliTestDirectory).Parent;
 
-            var config = new AnalysisConfig() { SonarScannerWorkingDirectory = "C:\\working" };
+            // Act
+            SonarScannerWrapper.FindScannerExe(logger, scannerCliTestDirectoryRoot.FullName, "0");
```

Comment:
could you please use a different version than `0`?

`TestUtils.CreateTestSpecificFolder` will create internal folders such as `sonar-scanner-msbuild\TestResults\Deploy_Andrei<FOO>\0` - where `0` is the result of `UniqueDirectory.CreateNext(testContext.TestDir);` 

having `0` as version as `0` as folder is confusing

maybe use a constant TestVersion = `99.99`?